spock-core/src/main/java/org/spockframework/compiler/condition/DefaultConditionRewriterResources.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.spockframework.compiler.condition;
+
+import org.codehaus.groovy.ast.ASTNode;
+import org.spockframework.compiler.AstNodeCache;
+import org.spockframework.compiler.ErrorReporter;
+import org.spockframework.compiler.IRewriteResources;
+import org.spockframework.compiler.SourceLookup;
+
+class DefaultConditionRewriterResources implements IRewriteResources {
+
+  private final AstNodeCache nodeCache;
+  private final SourceLookup lookup;
+  private final ErrorReporter errorReporter;
+  private final IConditionErrorRecorders errorRecorders;
+
+  DefaultConditionRewriterResources(
+    AstNodeCache nodeCache,
+    SourceLookup lookup,
+    ErrorReporter errorReporter,
+    IConditionErrorRecorders errorRecorders
+  ) {
+    this.nodeCache = nodeCache;
+    this.lookup = lookup;
+    this.errorReporter = errorReporter;
+    this.errorRecorders = errorRecorders;
```

Comment:
Please add `Objects.requireNonNull()` to indicate non-null params or add `@Nulable` annotations.

```suggestion
    this.nodeCache = requireNonNull(nodeCache);
    this.lookup = requireNonNull(lookup);
    this.errorReporter = requireNonNull(errorReporter);
    this.errorRecorders = requireNonNull(errorRecorders);
```

Reply:
For internal code it should be `org.spockframework.util.Assert#notNull(T)` and for user facing code it would be `org.spockframework.util.Checks#notNull`.

Reply:
Applied in [`461fb40` (#2112)](https://github.com/spockframework/spock/pull/2112/commits/461fb404837a959c6acb199a28036959a9a8e887)